mrp_subcontracting_partner_management/models/res_partner.py

```diff
@@ -6,258 +8,219 @@ class ResPartner(models.Model):
+    @api.model
+    def _update_name_translation(self, records, name):
+        """Update name field translation for records"""
+        self.env["ir.translation"].search(
+            [
+                ("name", "=", "{},name".format(records._name)),
+                ("res_id", "in", records.ids),
+                ("value", "!=", name),
+            ]
+        ).write({"value": name})
+
+    def _update_subcontractor_values_name(self, name):
         """
-        This Method is override to archive all subcontracting field
+        Update subcontractor related records:
+        - Location;
+        - Operation type;
+        - Route Rule for Subcontracting buy;
+        - Route Rule for Subcontracting resupply.
         """
-        for record in self:
-            record._set_subcontracting_values_active(False)
-        result = super(ResPartner, self).unlink()
-        return result
+        partners = self.filtered(lambda p: p.is_subcontractor_partner)
+        field_names = [*self.CHECK_DATA, "property_stock_subcontractor"]
+        field_names.remove("partner_picking_type_id")
+        for field in field_names:
+            records = partners.mapped(field)
+            records.write({"name": name})
+            self._update_name_translation(records, name)
+        type_name = "%s:  IN" % name
+        code = "".join(re.findall(r"\b\w", type_name))
+        picks = partners.mapped("partner_picking_type_id")
+        picks.write({"name": type_name, "sequence_code": code})
+        self._update_name_translation(picks, type_name)
 
-    def write(self, values):
-        for record in self:
-            is_subcontractor_partner = values.get("is_subcontractor_partner")
-            active = values.get("active")
-            if is_subcontractor_partner is not None:
-                values.update(record._update_subcontractor_entities_for_record(values))
-            if active is not None:
-                record._set_subcontracting_values_active(active)
-        super(ResPartner, self).write(values)
+    def unlink(self):
+        """This Method is override to archive all subcontracting field"""
+        self._set_subcontracting_values_active(False)
+        return super(ResPartner, self).unlink()
+
+    def write(self, vals):
+        if "is_subcontractor_partner" in vals:
+            self._update_subcontractor_entities_for_record(
+                vals.get("is_subcontractor_partner")
+            )
+        if "active" in vals:
+            self._set_subcontracting_values_active(vals.get("active"))
+        result = super(ResPartner, self).write(vals)
+        if vals.get("name"):
+            self._update_subcontractor_values_name(vals.get("name"))
+        return result
 
     @api.model
-    def create(self, values):
-        partner = super(ResPartner, self).create(values)
-        if values.get("is_subcontractor_partner", False):
-            partner._create_subcontractor_entities()
-        return partner
+    def create(self, vals):
```

Comment:
Do it create_multi for not losing performance in this critical object.

Reply:
Resolved. Added model_create_multi decorator and optimize a functions code.

Reply:
Please use correct nomenclature. The argument should be `vals_list`, and can't this be done inline for not having to create + write?

Reply:
@pedrobaeza After creation record created related records that is why used create + write.

Reply:
But that's the question: Why can't you create them before calling super, and pre-assign them in the dictionary values?

Reply:
@pedrobaeza Resolved.